keras/activations.py

```diff
@@ -4,14 +4,37 @@
 from .utils.generic_utils import deserialize_keras_object
 
 
-def softmax(x):
+def softmax(x, data_format=K.image_data_format()):
+    """
```

Comment:
Better way to do this is:
```python
def softmax(x, data_format=None):
    # ...
    elif ndim == 4:
        if not data_format:
            data_format = K.image_data_format()
        if data_format == 'channels_last':
            # ...
```
Both allow user to specify `data_format` and fallback to use `K.image_data_format()`